### PR TITLE
[FLINK-18971] Support to mount kerberos conf as ConfigMap and Keytab …

### PR DESCRIPTION
**Pull request by KarmaGYZ**
…as Secrete

## What is the purpose of the change

Support to mount kerberos conf as ConfigMap and Keytab as Secrete

## Brief change log

- Add "kubernetes.kerberos.krb5-conf.path" Config option, which specify the local location of the krb5.conf file to be mounted on the JobManager and TaskManager for Kerberos.
- Introduce KerberosMountDecorator

## Verifying this change

This change added tests and can be verified as follows:

- Existing UTs and IT cases.
- KerberosMountDecoratorTest

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs

### CODE REVIEW

File: flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/KerberosMountDecoratorTest.java
Comment (xintongsong) [inline flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/KerberosMountDecoratorTest.java:51]: Why is this class extends `KubernetesJobManagerTestBase`? It seems this test class can extends either `KubernetesJobManagerTestBase` or `KubernetesTaskManagerTestBase`, but cannot extends `KubernetesTestBase`. I think this is an indicator of bad abstraction. We may have another common base class for jobmanager/taskmanager test bases.

File: flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/KerberosMountDecoratorTest.java
Comment (xintongsong) [inline flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/KerberosMountDecoratorTest.java:123]: I think this test case is not needed. It is already covered by `testDecoratedFlinkContainer`. Despite it is required to mount the volumes on the pods before mounting them on the containers, from the contract we only cares about accessing the resources in container.

**Review by xintongsong — CHANGES_REQUESTED** [review]
@KarmaGYZ 
Thanks for opening this PR, and sorry for the late response.
The feature looks good to me. I have some comments on the implementation details. Please take a look.

**Comment by KarmaGYZ** [comment]
@flinkbot run azure

**Comment by KarmaGYZ** [comment]
@xintongsong Thanks for the review. Comments addressed.

**Review by xintongsong — APPROVED** [review]
Thanks for addressing my comment. LGTM.
I'll merge this after getting the ci green light.

**Pull request closed without merging**
